packages/core-blockchain/lib/blockchain.js

```diff
@@ -196,7 +196,8 @@ module.exports = class Blockchain {
       await deleteLastBlock()
     }
 
-    await this.database.deleteBlockCommit()
+    // Commit delete blocks
+    await this.database.commit()
```

Comment:
the only comment I have is maybe naming of commit() method, so it won't mislead to others that this is a general database commit. As it is relate to queuedQueries... maybe just rename this to `commitQueued()` or something similar.